### DISCUSSION TITLE
Built-in DNS Server only works for A and AAAA records, other types result in timeout

### DISCUSSION THREAD

**Discussion by helcoder** [XTLS/Xray-core#4048] [Q&A] [answered]
Hello everyone! I'm trying to setup Built-in DNS Server and everything is working perfectly except for non-A type requests. I know that only A and AAAA types are supported but as per documentation "[Other queries will not enter the built-in DNS server](https://xtls.github.io/en/config/dns.html#dns-server)" and I assumed that they should be proxied directly, however it doesn't seem to be the case. If I remove dns section from config and simply proxy port 53 then everything is working as expected, no timeouts whatsoever. Can something be done to avoid timeouts but still use Built-in DNS Server? I need it in order to use fakedns.

Config and logs are attached, however logs don't seem to contain any errors related to such requests.

[config.json](https://github.com/user-attachments/files/17846521/config.json)
[error.log](https://github.com/user-attachments/files/17846523/error.log)
[access.log](https://github.com/user-attachments/files/17846524/access.log)

**Comment by kingwilliam** [comment, marked as answer]
https://xtls.github.io/en/config/outbounds/dns.html

"nonIPQuery": "skip"

```
outbounds
        {
            "tag": "out-dns",
            "protocol": "dns",
            "settings": {
                "nonIPQuery": "skip"
            }
        }
```

**Reply by helcoder** [reply]
Lifesaver, works like a charm! Thank you!

**Reply by helcoder** [reply]
I have additional question, is it possible to route skipped requests via some outbound, for example shadowsocks? Can routing rules be applied to skipped requests?

**Reply by kingwilliam** [reply]
i'don't work or not, you need to try

concept:
if dns request is A or AAAA, outbound-dns will ignore 127.0.0.1:5353,
if dns request is not A or AAAA, will access 127.0.0.1:5353


```
inbound
{
    tag: "reroute-dns"
    protocol: "dokodemo-door",
    address: 127.0.0.1
    port: 5353
    settings: {
        address: 8.8.8.8
        port: 53
    }
}

route
{
   inbound: reroute-dns
   outbound: out-gooutside
}

outbound
   dns:
   {
      address: 127.0.0.1
      port: 5353
      nonipquery: skip
   },
   out-goutside
   {
      .....
   }
```

**Reply by helcoder** [reply]
Magic, this is working! Thank you again!
